src/customHooks/useLocalStorage.js

```diff
@@ -0,0 +1,14 @@
+import { useState, useEffect } from 'react'
+ 
+const retrievedFavs = JSON.parse(localStorage.getItem("favoriteCards"));
+
+export default function useLocalStorage(initialState) {
+    const [favorites, setFavorites] = useState(retrievedFavs ? retrievedFavs : initialState);
+
+    useEffect(() => {
+        const favsToBeStored = JSON.stringify(favorites);
+        localStorage.setItem("favoriteCards", favsToBeStored);
+      }, [favorites]);
+
+    return [favorites, setFavorites];
```

Comment:
We don't want this state exported, we want functions to get all of the favorites and set one favorite (hint: use id).